crates/sui-core/src/checkpoints/checkpoint_executor/mod.rs

```diff
@@ -440,7 +441,14 @@ async fn execute_transactions(
                 periods += 1;
             }
             Ok(Err(err)) => return Err(err),
-            Ok(Ok(_)) => {
+            Ok(Ok(effects)) => {
+                for (tx_digest, expected_effects_digest, actual_effects) in
+                    izip!(&all_tx_digests, &effects_digests, &effects)
+                {
+                    if expected_effects_digest != actual_effects.digest() {
+                        panic!("When executing checkpoint {checkpoint_sequence}, transaction {tx_digest} is expected to have effects {expected_effects_digest}, but got {}!", actual_effects.digest());
```

Comment:
```suggestion
                        panic!("When executing checkpoint {checkpoint_sequence}, transaction {tx_digest} is expected to have effects digest {expected_effects_digest}, but got {}!", actual_effects.digest());
```

Reply:
> Did this happen? Or just being proactively cautious?

Just being cautious since we did see inconsistent executions this week!